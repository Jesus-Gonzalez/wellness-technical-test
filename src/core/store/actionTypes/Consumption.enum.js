export const Consumption = {
  Fetch: 'FETCH_CONSUMPTIONS',
  FetchSuccess: 'FETCH_CONSUMPTIONS_SUCCESS',
  FetchError: 'FETCH_CONSUMPTIONS_ERROR'
}
