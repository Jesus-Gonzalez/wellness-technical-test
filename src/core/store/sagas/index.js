import { all } from 'redux-saga/effects'

import * as Consumption from './consumption'

export default function * RootSaga () {
  yield all([
    ...Object.values(Consumption).map(fn => fn())
  ])
}
