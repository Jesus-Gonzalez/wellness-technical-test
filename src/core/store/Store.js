import React from 'react'
import PropTypes from 'prop-types'
import { applyMiddleware, createStore, compose } from 'redux'
import { Provider } from 'react-redux'
import createSagaMiddleware from 'redux-saga'

import reducers from './reducers'
import sagas from './sagas'

const sagaMiddleware = createSagaMiddleware()

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const store = createStore(reducers, composeEnhancers(
  applyMiddleware(sagaMiddleware)
))

sagaMiddleware.run(sagas)

const propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.element
  ]).isRequired
}

export const Store = ({ children }) => {
  return (
    <Provider store={store}>
      {children}
    </Provider>
  )
}

Store.propTypes = propTypes
