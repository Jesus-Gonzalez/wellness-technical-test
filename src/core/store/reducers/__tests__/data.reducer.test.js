import { fetchConsumptions } from 'core/store/actions'
import * as Actions from 'core/store/actionTypes'

import reducer from '../data'

describe('redux', () => {
  describe('reducers', () => {
    it('data should be functional', () => {
      let state = reducer({}, { type: 'init' })
      expect(state).toMatchSnapshot()

      state = reducer(state, fetchConsumptions())
      expect(state).toMatchSnapshot()

      state = reducer(state, {
        type: Actions.Consumption.FetchSuccess,
        items: [{ id: 1 }, { id: 2 }]
      })
      expect(state).toMatchSnapshot()
    })
  })
})
