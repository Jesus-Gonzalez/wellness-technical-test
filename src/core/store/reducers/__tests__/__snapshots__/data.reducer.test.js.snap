// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux reducers data should be functional 1`] = `Object {}`;

exports[`redux reducers data should be functional 2`] = `
Object {
  "error": false,
  "loading": true,
  "status": "FETCH_CONSUMPTIONS",
}
`;

exports[`redux reducers data should be functional 3`] = `
Object {
  "error": false,
  "items": Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
  ],
  "loading": false,
  "status": "FETCH_CONSUMPTIONS_SUCCESS",
}
`;
